src/pages/messages/components/MessageList.tsx

```diff
@@ -1,22 +1,29 @@
-import { Box, Button, Divider, Skeleton, Typography } from '@mui/material';
+import { Box, Divider, Skeleton, Tooltip, Typography } from '@mui/material';
 import { useQuery } from '@tanstack/react-query';
-import { ReactNode, useState } from 'react';
+import { ReactNode } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
 import { fetchUser } from '../../../api/roleApi';
-import { ConfirmDialog } from '../../../components/ConfirmDialog';
 import { ErrorBoundary } from '../../../components/ErrorBoundary';
+import { RootState } from '../../../redux/store';
 import { Ticket } from '../../../types/publication_types/ticket.types';
 import { dataTestId } from '../../../utils/dataTestIds';
 import { toDateString } from '../../../utils/date-helpers';
-import { getFullName } from '../../../utils/user-helpers';
+import { useDeleteTicketMessage } from '../../../utils/hooks/useDeleteTicketMessage';
+import { getFullName, truncateName } from '../../../utils/user-helpers';
+import { MessageMenu } from './MessageMenu';
 import { ticketColor } from './TicketListItem';
 
 interface MessageListProps {
   ticket: Ticket;
+  refetchData?: () => void;
+  canDeleteMessage?: boolean;
 }
 
-export const TicketMessageList = ({ ticket }: MessageListProps) => {
+export const TicketMessageList = ({ ticket, refetchData, canDeleteMessage }: MessageListProps) => {
   const messages = ticket.messages ?? [];
+  const user = useSelector((store: RootState) => store.user);
+  const deleteTicketMessageMutation = useDeleteTicketMessage(ticket.id, refetchData);
```

Comment:
Denne brukes bare for å sendes som prop ned til `MessageMenu`. Sånn at den vil nå opprettes uansett om bruker kan slette eller ikke. Virker mye enklere å flytte det inn i den komponenten i stedet?